poetry/publishing/uploader.py

```diff
@@ -89,23 +89,27 @@ def files(self) -> List[Path]:
 
         return sorted(wheels + tars)
 
-    def auth(self, username: str, password: str) -> None:
+    def auth(self, username: Optional[str], password: Optional[str]) -> None:
         self._username = username
         self._password = password
 
     def make_session(self) -> requests.Session:
         session = requests.session()
-        if self.is_authenticated():
-            session.auth = (self._username, self._password)
+        auth = self.get_auth()
+        if auth is not None:
+            session.auth = auth
 
         session.headers["User-Agent"] = self.user_agent
         for scheme in ("http://", "https://"):
             session.mount(scheme, self.adapter)
 
         return session
 
-    def is_authenticated(self) -> bool:
-        return self._username is not None and self._password is not None
+    def get_auth(self) -> Optional[Tuple[str, str]]:
```

Comment:
This is a much more elegant construct, and `is_authenticated` was an ambiguous name. Nice.